dashboard/src/app/factories/load-factory/load-factory.controller.ts

```diff
@@ -244,7 +247,32 @@ export class LoadFactoryController {
           if (this.factory.id) {
             return this.factory.id === (w.attributes as any).factoryId;
           } else if (this.routeParams.url){
-            return this.routeParams.url === (w.attributes as any).factoryurl;
+            const factoryUrl = (w.attributes as any).factoryurl;
+            // compare factory URL and route params
+            if (angular.isDefined(factoryUrl)) {
+              const factoryUrlObj = new (window as any).URL(factoryUrl);
+              const pathIsCorrect = `${factoryUrlObj.origin}${factoryUrlObj.pathname}` === this.routeParams.url;
+              if (pathIsCorrect === false) {
+                return false;
+              }
+
+              let factoryUrlParamsNumber = 0;
+              let hasExtraKey = false;
+              factoryUrlObj.searchParams.forEach((volume: string, key: string) => {
```

Comment:
Up to you, but seems  `for .. of` loop is clearer here (and you may return just in place)

Reply:
ok